### PR TITLE
Upgrade from `python3.8` to `python3.10`

### PR DESCRIPTION
**Pull request by jumaffre**
This is part of the upgrade to Ubuntu 22.04 for SNP/virtual builds, where `python3.10` is available via `pip` by default. 

The couple of Ubuntu hosted CI executors are also upgraded to Ubuntu 22.04 so that `python3.10` is available by default.

### CODE REVIEW

**Comment by jumaffre** [comment]
The SNP pipeline is failing as expected as it is building the image off `main`. It will be fixed once the PR is merged.

File: .azure-pipelines-templates/deploy_aci.yml
Comment (jumaffre) [reply .azure-pipelines-templates/deploy_aci.yml:188]: This should have only been done on CI jobs that used Python (3.8 but now 3.10) but do not make use of any of our containers. In other words, it's not a user-facing change, but simply for us to use 22.04 for a few CI agents. I'll reflect that in the PR description.

**Review by eddyashton — APPROVED** [review]

**Comment by jumaffre** [comment]
This is currently blocked by the GitHub Pages job as Sphinx doesn't work with Python 3.10 because of https://github.com/python-babel/babel/issues/990. 

I see two options:
1. Wait for the fix (https://github.com/python-babel/babel/pull/1006) to be merged upstream (may take a while).
2. Temporarily relax the required version condition in `setup.py` so that `ccf` can be installed with `python3.8` (as there are no compatibility issues between 3.8 and 3.10).

File: .azure-pipelines-templates/checks.yml
Comment (achamayou) [inline .azure-pipelines-templates/checks.yml:12]: I think we ought to switch to calling `python3` everywhere, and communicate the version requirement through https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#python-requires

Wherever we can, using a shebang line or setuptools entry points would be nicer too, to avoid having to specify which  python, now that 2.x is thoroughly obsolete.

File: getting_started/setup_vm/roles/ccf_build/tasks/install.yml
Comment (achamayou) [inline getting_started/setup_vm/roles/ccf_build/tasks/install.yml:1]: I am confused by this, doesn't 22.04 ship with 3.10? If so, why do we need the ppa?

Comment (jumaffre) [reply getting_started/setup_vm/roles/ccf_build/tasks/install.yml:1]: This PR doesn't update the base container image to 22.04, and only updates Python to 3.10. Updating our container images to 22.04 will be a bigger change (in a separate PR) which cannot be completed just yet because of the lack of Open Enclave packages on 22.04.

**Comment by jumaffre** [comment]
As per the update on https://github.com/microsoft/CCF/issues/5291, we will not do this as there are no tangible benefits and we will remain on Ubuntu 20.04 for now.

**Pull request closed without merging**
